setup.py

```diff
@@ -1,23 +1,149 @@
+        if sysinfo.machine == 'x86_64':
+            platform = 'linux_x86_64'
+            return python_tag, abi_tag, platform
+        elif sysinfo.machine == 'aarch64':
+            raise ValueError(f"ivadomed doesn't handpick torch whls for {sysinfo.machine}, \
+            install torch==1.8.0+cpu, torchvision==0.9.0+cpu or \
+            torch==1.8.1+cu111, torchvision==0.9.1+cu111 from \
+            https://pytorch.org/get-started/previous-versions/")
+    elif sysinfo.system == 'Windows' and sysinfo.machine == 'AMD64':
+        platform = 'win_amd64'
+        return python_tag, abi_tag, platform
+    elif sysinfo.system == 'Darwin':
+        # TODO: update mac conditions as we upgrade to higher torch versions
+        # with separate binaries for 10 and 11 versions   
+        # macinfo = mac_ver()
+        # if macinfo[0].startswith('10') and macinfo[2] == 'x86_64':
+        platform = 'macosx_10_9_x86_64'
+        return python_tag, abi_tag, platform 
 
 
+def get_torch_whls(backend):
+    """Handpicks whls for torch and torchvision based on the backend.
+
+    Args:
+        backend (str): cpu or gpu
+
+    Returns:
+        list[str]: List of handpicked torch et al. whls
+    """
+    python_tag, abi_tag, platform = get_whl_info()
+    if backend == 'cpu':
+        if platform.startswith('macosx'):
+            _torch_whl = f'-{python_tag}-none-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{_torch_whl}'
+            _torchvision_whl = f'-{python_tag}-{abi_tag}-{platform}.whl'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/torchvision-0.9.0{_torchvision_whl}'
+            return torch_whl, torchvision_whl
+        else:
+            whl = f'%2B{backend}-{python_tag}-{abi_tag}-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{whl}'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/cpu/torchvision-0.9.0{whl}'
+            return torch_whl, torchvision_whl
+    elif backend == 'gpu':
+        if platform.startswith('macosx'):
+            # MacOS Binaries don't support CUDA, install from source if CUDA is needed
+            # install on CPU by default
+            _torch_whl = f'-{python_tag}-none-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cpu/torch-1.8.0{_torch_whl}'
+            _torchvision_whl = f'-{python_tag}-{abi_tag}-{platform}.whl'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/torchvision-0.9.0{_torchvision_whl}'
+            return torch_whl, torchvision_whl
+        else:
+            backend = 'cu111'
+            whl = f'%2B{backend}-{python_tag}-{abi_tag}-{platform}.whl'
+            torch_whl = f'torch@https://download.pytorch.org/whl/cu111/torch-1.8.1{whl}'
+            torchvision_whl = f'torchvision@https://download.pytorch.org/whl/cu111/torchvision-0.9.1{whl}'
```

Comment:
Was wondering why these are 1.8.1 while others are 1.8.0 (similar question for torchvision). Also, do you think we should potentially parameterize this number out such that if we decide to move all platform to 1.9 or other higher number, it should handle equally in the future across all platforms?

Reply:
Just replicated whatever versions we had currently 🙂